pages/api/token/temporary.ts

```diff
@@ -82,16 +82,43 @@ const createTemporaryToken = (email: string, tokenSecret: string): string => {
  * @param templateId
  * @returns the query result
  */
-const updateTemporaryToken = async (
-  temporaryToken: string,
-  email: string,
-  templateId: string
-): Promise<QueryResult> => {
-  return executeQuery(
-    await dbConnector(),
-    "UPDATE form_users SET temporary_token = ($1), updated_at = current_timestamp WHERE email = ($2) and template_id = ($3) and active = true",
-    [temporaryToken, email, templateId]
-  );
+const updateTemporaryToken = async (temporaryToken: string, email: string, templateId: string) => {
+  try {
+    // Throws an error is user is not found
+    const user = await prisma.formUser.findUnique({
+      where: {
+        templateId_email: {
+          email: email,
+          templateId: templateId,
+        },
+      },
+      select: {
+        email: true,
+        templateId: true,
+        active: true,
+      },
+    });
+    if (user?.active) {
+      await prisma.formUser.update({
+        where: {
+          templateId_email: {
+            email: user.email,
+            templateId: user.templateId,
+          },
+        },
+        data: {
+          temporaryToken: temporaryToken,
+        },
+      });
+    } else {
+      throw new Error("User is not found or not active");
```

Comment:
Do these type of API errors need to also be in FR?

Reply:
Not that I'm aware of.  API errors are usually not internationalized which is why HTTP status codes are predominantly used.